packages/grafana-runtime/src/datasources/DataSourcePicker.tsx

```diff
@@ -2,13 +2,12 @@
 import React, { PureComponent } from 'react';
 
 // Components
-import { HorizontalGroup, Select } from '@grafana/ui';
-import { DataSourceInstanceSettings, SelectableValue } from '@grafana/data';
+import { HorizontalGroup, PluginSignatureBadge, Select } from '@grafana/ui';
+import { DataSourceInstanceSettings, isUnsignedPluginSignature, SelectableValue } from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
-import { isUnsignedPluginSignature, PluginSignatureBadge } from '../../../features/plugins/PluginSignatureBadge';
-import { getDataSourceSrv } from '@grafana/runtime';
+import { getDataSourceSrv } from '../services/dataSourceSrv';
```

Comment:
I assume this is the one blocking from moving to grafana/ui. Maybe DataSourcePicker should have a property dependency on the data source service?

Reply:
Yeah, but that doesn't help because the interface for the DataSourceSrv is in grafana/runtime and it has a dependency on grafana/ui. If it was located in grafana/data that would have been possible, I guess.

Reply:
Alternatively, instead of moving this to runtime, we could create a registry of runtime-dependant components, similar to standardEditorsRegistry. This would allow us to surface core components to plugins, but we would lose type checking.on those components retrieved from such a registry.

Reply:
hmm, yeah - I thought a bit on that yesterday as well but doesn't feel optimal.

Reply:
I am happy to change this if we can come up with a better way of doing it.

Reply:
@mckn been thinking about this a bit. Since this is the first time we actually want a runtime-dependant ui component being shared with plugins I will agree with you and @torkelo to keep the DataSourcePicker in the runtime. I think this requires us to externalize react dependency in the runtime package, otherwise react import in this component will make react land in the runtime's bundle.

Reply:
Awsome, I will add it to the external list.